benches/all.rs

```diff
@@ -7,58 +10,116 @@ use primes::{
     Primes,
 };
 
-// The largest prime number we will search for for eratosthenes-based algorithms.
-const LARGEST_PRIME_TO_SEARCH: u64 = 821641;
-
-// The index of LARGEST_PRIME_TO_SEARCH in an array of prime numbers.
-const LARGEST_PRIME_INDEX: usize = 65536;
-
-fn live_first_primes(c: &mut Criterion) {
-    c.bench_function("live_first_primes", |b| {
-        b.iter(|| {
-            let _: Vec<u64> = LiveFirstPrimes::new().take(LARGEST_PRIME_INDEX).collect();
-        })
-    });
+struct Input {
+    largest_prime: u64,
+    largest_prime_index: usize,
+    search_space: u64,
 }
 
-fn live_second_primes(c: &mut Criterion) {
-    c.bench_function("live_second_primes", |b| {
-        b.iter(|| {
-            let _: Vec<u64> = LiveSecondPrimes::new().take(LARGEST_PRIME_INDEX).collect();
-        })
-    });
-}
+impl Input {
+    fn new(search_space: u64) -> Self {
+        let primes = skip_2::primes(search_space);
 
-fn sieve_og(c: &mut Criterion) {
-    c.bench_function("sieve_og", |b| {
-        b.iter(|| {
-            let _ = og::primes(LARGEST_PRIME_TO_SEARCH);
-        })
-    });
+        Self {
+            largest_prime: *primes.last().unwrap(),
+            largest_prime_index: primes.len(),
+            search_space,
+        }
+    }
 }
 
-fn sieve_skip_2(c: &mut Criterion) {
-    c.bench_function("sieve_skip_2", |b| {
-        b.iter(|| {
-            let _ = skip_2::primes(LARGEST_PRIME_TO_SEARCH);
-        })
-    });
+impl fmt::Display for Input {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.search_space)
+    }
 }
 
-fn test_division(c: &mut Criterion) {
-    c.bench_function("test_division", |b| {
-        b.iter(|| {
-            let _: Vec<u32> = Primes::new().take(LARGEST_PRIME_INDEX).collect();
-        })
-    });
+fn bench_em(c: &mut Criterion) {
+    let mut group = c.benchmark_group("Primes");
+    // For each tuple, the first number is the largest prime number we will search for for
```

Comment:
Is this comment still applicable? It mentions tuples, which I don't see.

Reply:
I think this comment is pre-`Input` so might be better phrased as "Each number in this list defines a search space in which we will record the largest prime and the index of that prime" or similar